sled-agent/src/support_bundle/storage.rs

```diff
@@ -378,7 +412,9 @@ impl<'a> SupportBundleManager<'a> {
     }
 
     // Returns a dataset that the sled has been explicitly configured to use.
-    async fn get_configured_dataset(
+    //
+    // Returns an error if this dataset is not mounted.
```

Comment:
This is used to access the "parent dataset" on which the "support bundle dataset" gets placed.

E.g., "Look up the `crypt/debug` dataset by UUIDs".

I'm adding these checks to also validate: ALSO confirm that this dataset is mounted.

This is tested in `cannot_create_bundle_on_unmounted_parent`